src/components/pages/teacher-info-page/TeacherInformPage.jsx

```diff
@@ -0,0 +1,284 @@
+import React, { useEffect, useState } from 'react';
+import { getUserData, getClassesData } from '../../../util/DataBaseRequests';
+import { useAuth } from '../../../AuthProvider';
+import { useParams } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
+import Loader from '../../common/Loader';
+
+const TeacherInfoPage = () => {
+  const { userData } = useAuth();
+  const { teacherId } = useParams();
+  const navigate = useNavigate();
+  const [teacherInfo, setTeacherInfo] = useState(null);
+  const [classes, setClasses] = useState([]);
+  const [classesError, setClassesError] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedImage, setSelectedImage] = useState(null);
+
+  useEffect(() => {
+    const fetchTeacherAndClasses = async () => {
```

Comment:
Get `getUserData` and `getClassesData` is lumped into one catch-all block, making it difficult to discern whether an error occurred while fetching the teacher data or the classes data. Ideally you want each API call to have separate error handling.